src/ethereum/frontier/vm/instructions/block.py

```diff
@@ -36,15 +36,13 @@ def block_hash(evm: Evm) -> None:
     OutOfGasError
         If `evm.gas_left` is less than `20`.
     """
-    # TODO: This function hasn't been tested. Need to come up with unit tests
-    # for this function.
     evm.gas_left = subtract_gas(evm.gas_left, GAS_EXTERNAL)
 
-    block_number = pop(evm.stack)
+    block_number = int(pop(evm.stack))
     # The block age (1-indexed) with respect to the current block number.
     # A block having age of `a` indicates that it is `a` blocks
     # behind the current block number.
-    block_age = evm.env.number - block_number
+    block_age = int(evm.env.number) - block_number
 
     if block_age < 1 or block_age > 256:
```

Comment:
Would it be possible to write this without negative integers, maybe `if block_number + 1 > evm.env.number`?

Reply:
can you check the latest commit? Not sure if that complicates things.